onnxruntime/contrib_ops/cpu/bert/attention_cpu_base.h

```diff
@@ -227,10 +240,23 @@ class AttentionCPUBase : public AttentionBase {
       present += SafeInt<ptrdiff_t>(batch_size) * num_heads_ * total_sequence_length * v_head_size;
     }
 
-    const double cost =
-        static_cast<double>(sequence_length) * static_cast<double>(v_head_size) * static_cast<double>(sequence_length);
+    // The cost of Gemm
+    TensorOpCost unit_cost;
+    unit_cost.compute_cycles = static_cast<double>(2 * sequence_length * v_head_size * total_sequence_length);
+    unit_cost.bytes_loaded = double(sequence_length * total_sequence_length * sizeof(T) + v_head_size * total_sequence_length * sizeof(T));
+    unit_cost.bytes_stored = double(sequence_length * v_head_size * sizeof(T));
+
+    if (present || present_value) {
+      size_t bytes_to_copy_value = sizeof(T) * size_t(past || past_value ? present_chunk_length : present_chunk_length - past_chunk_length);
```

Comment:
Need copy the whole present chunk length since the code copies past first:
https://github.com/microsoft/onnxruntime/blob/9d06e1bfa469bce92d7aa2ebfa36c7a9714ba119/onnxruntime/contrib_ops/cpu/bert/attention_helper.h#L159-L175

Reply:
if (past || past_value) is true, it copies the whole present.

Reply:
Right.  `size_t bytes_to_copy_value = sizeof(T) * size_t(present_chunk_length)`